buf/registry/module/v1beta1/commit_service.proto

```diff
@@ -35,10 +35,13 @@ service CommitService {
   rpc ListCommitHistory(ListCommitHistoryRequest) returns (ListCommitHistoryResponse) {
     option idempotency_level = NO_SIDE_EFFECTS;
   }
-  // Create commits on a Module with associated Content.
+  // Create commits on a Module with associated content (e.g. blobs, branches, tags).
   //
   // This is used by push and sync.
   //
+  // This operation is atomic. Either all Commits and associated content are created, or an error happens and none of
+  // them are created.
```

Comment:
@saquibmian @unmultimedio @doriable We need to document the atomicity of this request. In this PR I am presuming that this is atomic, but want to verify with you that doing this atomically is both possible and desirable. Can you confirm?

Reply:
Yes, and yes.

Reply:
Thanks!